tmpfs/tmpfs.go

```diff
@@ -0,0 +1,120 @@
+package tmpfs
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-billy.v2/subdirfs"
+)
+
+type tmpFs struct {
+	fs        billy.Filesystem
+	tmp       billy.Filesystem
+	tempFiles map[string]bool
```

Comment:
make this attribute a `map[string]billy.Filesystem`, call it `resolver`, add add a `resolve` method instead of `isTempFile`.  This should make the code even simpler I believe.